//Define the used version of proto
syntax = "proto3";

message IoTSample {
  float pm1 = 1;
  float pm25 = 2;
  float pm10 = 3;
  string name = 4;
}

message DataSample {
  repeated IoTSample iot_data = 1;
  float air_temperature = 2;
  float relative_humidity = 3;
  float precipitation = 4;
  float air_pressure = 5;
  float wind_speed = 6;
  float wind_direction = 7;
}

message ResponseSample {
  float raw_pm1 = 1;
  float raw_pm25 = 2;
  float raw_pm10 = 3;
  float calibrated_pm25 = 4;
  float calibrated_pm10 = 5;
  string sensor_name = 6;
}

message CalibResponse {
  repeated ResponseSample data = 1;
}

message Empty {
}

message InputSample {
  float raw_pm1 = 1;
  float raw_pm25 = 2;
  float raw_pm10 = 3;
  float calibrated_pm25 = 4;
  float calibrated_pm10 = 5;
  string sensor_name = 6;
}

message InputData {
  repeated InputSample data = 1;
}

service AQDataSource {
  rpc request_update(Empty) returns (DataSample);
}

service Calibration {
  rpc calibrate_sample(DataSample) returns (CalibResponse);
}

service Visualization {
  rpc set_values(InputData) returns (Empty);
}